From e879f68a77182a30e0ee2c1ffaea9ae9cd12f326 Mon Sep 17 00:00:00 2001
Message-Id: <e879f68a77182a30e0ee2c1ffaea9ae9cd12f326.1241177871.git.joerg@alea.gnuu.de>
From: =?utf-8?q?J=C3=B6rg=20Sommer?= <joerg@alea.gnuu.de>
Date: Fri, 1 May 2009 13:17:26 +0200
Subject: [PATCH 1/4] main: Remove locale setting when querying dpkg
MIME-Version: 1.0
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: 8bit

The output of dpkg is localized, which makes the match for “diversion by”
fail. Call dpkg with an environment where LC_ALL is set to C, the POSIX
locale.

% LC_ALL=de_DE.UTF-8 dpkg -S /usr/bin/mutt
Umleitung durch mutt-patched von: /usr/bin/mutt
Umleitung durch mutt-patched zu: /usr/bin/mutt.org
mutt-patched, mutt: /usr/bin/mutt
% LC_ALL=C dpkg -S /usr/bin/mutt
diversion by mutt-patched from: /usr/bin/mutt
diversion by mutt-patched to: /usr/bin/mutt.org
mutt-patched, mutt: /usr/bin/mutt
---
 checkrestart |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/checkrestart b/checkrestart
index 2834c50..bbea0c8 100755
--- a/checkrestart
+++ b/checkrestart
@@ -127,11 +127,14 @@ def main():
             print "Process %s (PID: %d) "  % (process.program, process.pid)
             process.listDeleted()
 
+    lc_all_c_env = os.environ
+    lc_all_c_env['LC_ALL'] = 'C'
 
     packages = {}
     diverted = None
     dpkgQuery = ["dpkg", "--search"] + programs.keys()
-    dpkgProc = subprocess.Popen(dpkgQuery, stdout=subprocess.PIPE, stderr=None)
+    dpkgProc = subprocess.Popen(dpkgQuery, stdout=subprocess.PIPE, stderr=None,
+                                env = lc_all_c_env)
     for line in dpkgProc.stdout.readlines():
         if line.startswith('local diversion'):
             continue
-- 
1.6.3.rc1

